exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/TransactionScope.kt

```diff
@@ -5,27 +5,20 @@ import org.jetbrains.exposed.sql.Transaction
 import kotlin.properties.ReadWriteProperty
 import kotlin.reflect.KProperty
 
-@Suppress("UNCHECKED_CAST")
-fun <T:Any> transactionScope(init: Transaction.() -> T) = TransactionStore(init) as ReadWriteProperty<Any?, T>
-fun <T:Any> nullableTransactionScope() = TransactionStore<T>()
-
-class TransactionStore<T:Any>(val init: (Transaction.() -> T)? = null) : ReadWriteProperty<Any?, T?> {
+fun <T: Any> transactionScope(init: Transaction.() -> T): ReadWriteProperty<Any?, T> = TransactionScope(init)
+fun <T: Any> nullableTransactionScope(init: Transaction.() -> T? = { null }): ReadWriteProperty<Any?, T?> = TransactionScope(init)
```

Comment:
@Tapac `nullableTransactionScope` is currently unused within the library.  Is it worth deprecating?  `transactionScope` could be opened up to allow nullable values i.e. `transactionScope<Int>(...)` vs `transactionScope<Int?>(..)`.